test/MOI_wrapper.jl

```diff
@@ -1,73 +1,117 @@
-                "number_threads",
-                # `TimeLimitSec` not supported.
-                "time_limit_sec",
-                # ArgumentError: The number of constraints in SCSModel must be greater than 0
-                "solve_unbounded_model",
-                # Integer and ZeroOne sets are not supported
-                "solve_integer_edge_cases",
-                "solve_objbound_edge_cases",
-                "solve_zero_one_with_bounds_1",
-                "solve_zero_one_with_bounds_2",
-                "solve_zero_one_with_bounds_3",
-            ],
-        )
-    end
+const MOI = MathOptInterface
 
-    @testset "Continuous linear problems with $T" begin
-        MOIT.contlineartest(bridged, config)
+function runtests()
+    for name in names(@__MODULE__; all = true)
+        if startswith("$(name)", "test_")
+            @testset "$(name)" begin
+                getfield(@__MODULE__, name)()
+            end
+        end
     end
+    return
+end
 
-    @testset "ADMMIterations attribute with $T" begin
-        MOIT.linear1test(bridged, config)
-        @test MOI.get(bridged, SCS.ADMMIterations()) > 0
-    end
+test_DirectSolver() = _test_runtests(SCS.DirectSolver)
 
-    @testset "Continuous quadratic problems with $T" begin
-        MOIT.qcptest(bridged, config)
-    end
+test_IndirectSolver() = _test_runtests(SCS.IndirectSolver)
 
-    @testset "Continuous conic problems with $T" begin
-        MOIT.contconictest(bridged, config, ["rootdets", "logdets"])
-    end
+function _test_runtests(linear_solver)
+    optimizer = SCS.Optimizer()
+    MOI.set(
+        optimizer,
+        MOI.RawOptimizerAttribute("linear_solver"),
+        linear_solver,
+    )
+    MOI.set(optimizer, MOI.RawOptimizerAttribute("eps"), 1e-6)
+    MOI.set(optimizer, MOI.Silent(), true)
+    model = MOI.Bridges.full_bridge_optimizer(
+        MOI.Utilities.CachingOptimizer(
+            MOI.Utilities.UniversalFallback(MOI.Utilities.Model{Float64}()),
+            optimizer,
+        ),
+        Float64,
+    )
+    MOI.Test.runtests(
+        model,
+        MOI.Test.Config(
+            atol = 1e-4,
+            exclude = Any[
+                MOI.ConstraintBasisStatus,
+                MOI.VariableBasisStatus,
+                MOI.ConstraintName,
+                MOI.VariableName,
+                MOI.ObjectiveBound,
+            ],
+        ),
+        exclude = String[
+            # Unexpected test failures:
+            #   Failures getting ConstraintFunction
+            "test_basic_ScalarQuadraticFunction_LessThan",
+            "test_basic_VectorOfVariables_GeometricMeanCone",
+            "test_conic_SecondOrderCone_negative_initial_bound",
+            "test_model_ModelFilter_AbstractConstraintAttribute",
+            "test_model_copy_to_UnsupportedAttribute",
+            "test_modification_set_function_single_variable",
+            "test_objective_incorrect_modifications",
+            "test_solve_optimize_twice",
```

Comment:
Plenty of failures here to debug. A lot of these look upstream. For example the solve objective twice one is due to our `copy_to_and_optimize!`. Obviously something isn't being set right in the caching optimizer.